src/vfs2.h

```diff
@@ -181,4 +181,13 @@ int vfs2_pseudo_read_end(sqlite3_file *file, unsigned i);
  */
 void vfs2_destroy(sqlite3_vfs *vfs);
 
+/**
+ * Declare a relationship between two databases opened by (possibly distinct)
+ * vfs2 instances.
+ *
+ * Intended for use by unit tests. The arguments are main file handles
+ * (SQLITE_FCNTL_FILE_POINTER).
+ */
+void vfs2_ut_sm_relate(sqlite3_file *orig, sqlite3_file *targ);
```

Comment:
I'm reluctant to move our `struct file` or `struct entry` into vfs2.h---it seems good to maintain the conciseness of that file and keep all the implementation details of vfs2 in one place. As for making the function uncallable outside tests, I agree it would be nice to be able to ifdef out bits of code so that they only exist if dqlite is being built for testing (and this is a good candidate). We'd need to tinker with the build system to implement this, because right now we don't (always) compile source files separately for the installable artifacts and the tests.

I think a good local improvement for this function specifically would be to just expose an accessor for the wtx sm, which will be useful also outside the unit tests (e.g. we'll want to sm_relate it to other state machines related to write transactions). What do you think?